packages/toolpad-app/src/components/AppEditor/index.tsx

```diff
@@ -42,14 +44,39 @@ interface FileEditorProps {
 }
 
 function FileEditor({ appId }: FileEditorProps) {
+  const dom = useDom();
+  const app = appDom.getApp(dom);
+  const { pages = [] } = appDom.getChildNodes(dom, app);
+  const firstPage = pages[0];
+
+  const hasPages = React.useMemo(() => {
+    if (firstPage?.id) {
+      return true;
+    }
+    return false;
+  }, [firstPage]);
+
   return (
     <Routes>
       <Route element={<AppEditorShell appId={appId} />}>
         <Route path="connections/:nodeId" element={<ConnectionEditor appId={appId} />} />
         <Route path="apis/:nodeId" element={<ApiEditor appId={appId} />} />
         <Route path="pages/:nodeId" element={<PageEditor appId={appId} />} />
         <Route path="codeComponents/:nodeId" element={<CodeComponentEditor appId={appId} />} />
-        <Route path="*" element={null} />
+        <Route path="codeComponents/:nodeId" element={<CodeComponentEditor appId={appId} />} />
+        <Route
+          index
+          element={
+            hasPages ? (
+              <Navigate to={`pages/${firstPage.id}`} />
+            ) : (
+              <NotFoundEditor
```

Comment:
Perhaps we can make a page that has the text "This app has no pages." and a button underneath that says "create page" that opens the page creation dialog?

Reply:
Does something like this make sense? 
<img width="1511" alt="Screenshot 2022-05-24 at 6 37 17 PM" src="https://user-images.githubusercontent.com/19550456/170042172-1ec85096-07c1-4823-965a-dcd9af2a5fc6.png">

Intentionally trying to avoid the pattern of one line text and a button centred on the page

Reply:
> Intentionally trying to avoid the pattern of one line text and a button centred on the page

I don't see a problem with that. I don't think `Alert` is the right paradigm for an empty state.

Reply:
<img width="1512" alt="Screenshot 2022-05-25 at 5 13 55 PM" src="https://user-images.githubusercontent.com/19550456/170254419-b6545b33-54b4-45fa-974f-9f2d9dcb1527.png">